### ISSUE TITLE
There's a lot of duplicate code

### ISSUE THREAD

**Issue by zhangruiyu** [pinchbv/floor#746]
![image](https://user-images.githubusercontent.com/10591822/224218362-716ad0be-d3c4-4fe4-be68-57e4f76f0f1a.png)

the extraction method would have been better

![image](https://user-images.githubusercontent.com/10591822/224218470-2d22af2b-3b38-4101-8296-55ecf8a69098.png)

**Comment by dkaera** [comment]
Yes, you're right. The database generator relies on pre-established patterns, which is why we're seeing these duplicates. Although we can enhance it in the future, it's more of an aesthetic concern rather than a performance issue. So, it's not a top priority at the moment. Nonetheless, I appreciate you bringing it to our attention.
